src/components/PostBody.js

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback } from 'react';
+import { navigate, StaticQuery, graphql } from 'gatsby';
+
+export default function PostBodyWrapper({ html }) {
+  return (
+    <StaticQuery
+      query={graphql`
+        query GetLanguages {
+          allMarkdownRemark(
+            filter: { fields: { slug: { regex: "/^/[^/]+/$/" } } }
+          ) {
+            edges {
+              node {
+                fields {
+                  slug
+                }
+                frontmatter {
+                  langs
+                }
+              }
+            }
+          }
+        }
+      `}
+      render={data => (
+        <PostBody
+          html={html}
+          posts={data.allMarkdownRemark.edges.map(edge => ({
+            slug: edge.node.fields.slug,
+            langs: edge.node.frontmatter.langs,
+          }))}
+        />
+      )}
+    />
+  );
+}
+
+function PostBody({ html, posts }) {
+  const handler = useCallback(event => {
+    if (
+      event.target.matches('a[href]') &&
+      event.target.host === location.host
+    ) {
+      const dest = event.target.pathname;
+      const src = location.pathname;
+      if (src.match(/^\/[\w-]+\/.+\//)) {
+        // localized page
+        const [, lang] = /^\/([\w-]+)\//.exec(src);
+        const post = posts.find(post => post.slug === dest);
```

Comment:
Switched to a slightly (perhaps?) more performant version that doesn’t create a new array on each render and uses an object instead of an array for lookups.